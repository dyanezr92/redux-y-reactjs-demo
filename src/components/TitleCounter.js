import React from 'react'
import PropTypes from 'prop-types'

const TitleCounter = ({ numElements, elementsName }) => (
  <h3 className="text-center">({ numElements } { elementsName })</h3>
);

TitleCounter.propTypes = {
  numElements: PropTypes.number.isRequired,
  elementsName: PropTypes.string.isRequired
}

export default TitleCounter
