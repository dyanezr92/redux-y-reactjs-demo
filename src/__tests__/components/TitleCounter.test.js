import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

import TitleCounter from '../../components/TitleCounter';

describe('<TitleCounter numElements={ 5 } elementsName="comments"/>', () => {
  describe('render()', () => {
    test('renders the title counter', () => {
      const wrapper = shallow(<TitleCounter numElements={ 5 } elementsName="comments"/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
