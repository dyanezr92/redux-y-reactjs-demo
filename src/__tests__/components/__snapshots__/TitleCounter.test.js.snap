// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleCounter numElements={ 5 } elementsName="comments"/> render() renders the title counter 1`] = `
<h3
  className="text-center"
>
  (
  5
   
  comments
  )
</h3>
`;
